### ISSUE TITLE
Record definition style

### ISSUE THREAD

**Issue by matthid** [dotnet/docs#6775]
There are at least 3 ways to define a record:

```fsharp
type Record =
  { clients: int
    field: string }
    member x.MyField = x.field
```

```fsharp
type Record =
    { clients: int
      field: string }
    member x.MyField = x.field
```

```fsharp
type Record = {
    clients: int
    field: string 
} with
    member x.MyField = x.field
```

```fsharp
type Record =
    {
        clients: int
        field: string 
    }
    member x.MyField = x.field
```

If we cann't recommend a particular style (because people do it differently) we should at least list the common ones to help new-comers (and list advantages and disadvantages...

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8c09996f-24ac-cf39-ef43-d7909cbc2b89
* Version Independent ID: 1efdd523-8552-6421-e149-74f76826ef8f
* Content: [F# code formatting guidelines](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting)
* Content Source: [docs/fsharp/style-guide/formatting.md](https://github.com/dotnet/docs/blob/master/docs/fsharp/style-guide/formatting.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **phcart**

**Comment by cartermp** [comment]
The current guidance does not enumerate the various ways to define records, though I'd be open to a PR that does lay out a few of these other styles.

**Comment by matthid** [comment]
@cartermp Should we discuss and decide on one of the above with some people or just list those three (personally, I don't really like all of them) ?

**Comment by vasily-kirichenko** [comment]
```fsharp
type Record =
  { Clients: int
    Field: string }
  member x.MyField = x.field
```

**Comment by 7sharp9** [comment]
The problem is none of them have a particularly nice line up between the end of definition and a member.  Having the requirement to use  `{ }` seems odd.

**Comment by matthid** [comment]
@7sharp9 Note that while I completely agree with you that any of the styles "feel" awkward this issue is not really about that. What I'm asking for is: What do we suggest newcomers to use considering the current state of the language.

I think your comment should be a language suggestion or discussed with @dsyme directly ;)

**Comment by matthid** [comment]
@vasily-kirichenko Now I realize that your suggestion is slightly different:

```fsharp
type Record =
<indent  >{ Clients: int
<indent+2>Field: string }
<indent  >member x.MyField = x.field
```

vs

```fsharp
type Record =
<indent-2>{ Clients: int
<indent  >Field: string }
<indent  >member x.MyField = x.field
```

which was my suggestion. It seems your suggestion is also what the compiler uses and looks quite good. Maybe we should just go with that?

**Comment by vasily-kirichenko** [comment]
@matthid I totally agree :)

**Comment by rodrigovidal** [comment]
I use the same style that @vasily-kirichenko does.

**Comment by cartermp** [comment]
One thing to keep in mind about using what the compiler does is documented here: #5552

TL;DR, for beginners who don't know how to please the compiler with indent/outdent rules, this is the best way to do it without getting yelled at by an IDE:

```fsharp
type MyRecord =
    {
        SomeField : int
    }
    interface IMyInterface
```

@vasily-kirichenko's suggestion is what you'll likely find the most online, which is a virtue. But so is recommending something that is as resilient as possible. The current section shows neither, so I think either of these is an improvement.

**Comment by matthid** [comment]
@cartermp Yes good feedback and indeed something we want to fix. The syntax you suggest seems the most F# "untypical" to me as it uses a lot of "blank lines" and noise of all the suggestions. Shouldn't we try to improve our tooling instead?

**Comment by matthid** [comment]
We could also go:

```fsharp
type Record =
    {   clients: int
        field: string }
    member x.MyField = x.field
```

```fsharp
type Record =
<i  >{<i-1>clients: int
<i  ><i   >field: string }
<i  >member x.MyField = x.field
```

which I guess would fix this problem as well. But this is never seen anywhere at the moment (or at least I have never seen it).

**Comment by AnthonyLloyd** [comment]
3 means cleaner git diffs while saving vertical and horizontal space.

**Comment by AnthonyLloyd** [comment]
3's instances are consistent:

```fsharp
let empty = {
    Field1 : int
    Field2 : string
}
```

can also be consistent with list / array style:

```fsharp
div [
    firstThing prop1 [
        innerThing prop2 prop3
        innerThing prop4 prop4
    ]
    secondThing prop5 [
        lastThing prop6 []
    ]
] 
```

**Comment by matthid** [comment]
Added a suggestion PR https://github.com/dotnet/docs/pull/6956, which tries to capture this discussion
